File: backend/src/main/java/com/ssafy/smartstore/service/FavoriteService.java
package com.ssafy.smartstore.service;

import java.util.List;
import java.util.stream.Collectors;

import org.springframework.stereotype.Service;

import com.ssafy.smartstore.dto.FavoriteProduct;
import com.ssafy.smartstore.repository.FavoriteProductRepository;

@Service
public class FavoriteService {

    private final FavoriteProductRepository favoriteRepo;

    public FavoriteService(FavoriteProductRepository favoriteRepo) {
        this.favoriteRepo = favoriteRepo;
    }

    public void addFavorite(String userId, int productId) {
        if (!favoriteRepo.findByUserIdAndProductId(userId, productId).isPresent()) {
            FavoriteProduct favorite = new FavoriteProduct();
            favorite.setUserId(userId);
            favorite.setProductId(productId);
            favoriteRepo.save(favorite);
        }
    }

    public void removeFavorite(String userId, int productId) {
        favoriteRepo.findByUserIdAndProductId(userId, productId)
                .ifPresent(favoriteRepo::delete);
    }

    public List<Integer> getFavorites(String userId) {
        return favoriteRepo.findByUserId(userId)
                .stream()
                .map(FavoriteProduct::getProductId)
                .collect(Collectors.toList());
    }
}

File: app/src/main/java/com/ssafy/finalpass/service/FavoriteService.kt
package com.ssafy.finalpass.service

import com.ssafy.finalpass.dto.FavoriteRequest
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.HTTP
import retrofit2.http.POST
import retrofit2.http.Path

interface FavoriteService {

    // 찜 추가
    @POST("api/favorites")
    suspend fun addFavorite(@Body req: FavoriteRequest): Response<Void>

    // 찜 삭제
    @HTTP(method = "DELETE", path = "api/favorites", hasBody = true)
    suspend fun removeFavorite(@Body req: FavoriteRequest): Response<Void>

    // 사용자 찜 목록 조회 (productId 리스트 반환)
    @GET("api/favorites/{userId}")
    suspend fun getFavorites(@Path("userId") userId: String): Response<List<Int>>
}